src/common/Input/Input.js

```diff
@@ -37,52 +38,73 @@ const Input = React.forwardRef(
     ref
   ) => {
     const [inputIsFocused, setInputIsFocused] = useState(false)
+    const [typedValue, setTypedValue] = useState(value ?? '')
     const input = React.createRef()
     const inputClassNames = classnames(
       'input',
       className,
-      (inputIsFocused || placeholder) && floatingLabel && 'active-input',
+      (inputIsFocused || placeholder || typedValue.length > 0) &&
+        floatingLabel &&
+        'active-input',
       required && 'input_required'
     )
     const labelClassNames = classnames(
       'input__label',
       floatingLabel && 'input__label-floating',
-      (inputIsFocused || placeholder) && floatingLabel && 'active-label',
+      (inputIsFocused || placeholder || typedValue.length > 0) &&
+        floatingLabel &&
+        'active-label',
       infoLabel && 'input__label_info'
     )
     const wrapperClassNames = classnames(wrapperClassName, 'input-wrapper')
 
     useEffect(() => {
-      if (input.current.value.length > 0) {
-        setInputIsFocused(true)
-      }
-
       if (focused) {
         input.current.focus()
+        setInputIsFocused(true)
       }
     }, [input, focused])
 
-    const handleClick = event => {
-      if (event.target.value.length > 0) {
-        setInputIsFocused(true)
-      } else {
+    const matchOnClick = item => {
+      setTypedValue(item)
+      setInputIsFocused(false)
+      onChange(item)
+    }
+
+    const onInputBlur = event => {
+      if (
+        !event.relatedTarget ||
+        !event.relatedTarget?.closest('.suggestion-list')
+      ) {
         setInputIsFocused(false)
+
+        if (onBlur) {
+          onBlur()
+        }
```

Comment:
I say either do:
```suggestion
        onBlur?.()
```
Or:
```suggestion
        if (typeof onBlur 'function') {
          onBlur()
        }
```
Or we can give a no-op function (`() => {}`) as the default value foe `onBlur` and just do:
```suggestion
        onBlur()
```